### ISSUE TITLE
Git import/preview

### ISSUE THREAD

**Issue by kylemcdonald** [processing/p5.js-web-editor#753]
I have a new feature request :)

I just taught a workshop with the p5.js editor. Usually I teach this workshop in 6 hours but the editor made it possible for me to teach it in 3 hours, due to the quicker setup time. It makes a huge difference.

The big difficulty in preparation for me was in uploading all my content to the editor.

Here are the examples I wanted to upload: https://github.com/kylemcdonald/cv-examples

For example, on this page if you click "edit" then you can see it links to the editor: https://kylemcdonald.github.io/cv-examples/AverageBrightness/

What would be amazing is to be able to simply point to:

https://editor.p5js.org/git/https://github.com/kylemcdonald/cv-examples.git/AverageBrightness

And have the editor know that it needs to fetch the repo and check out the specific folder.

This way the workshop/class materials could be kept up to date on GitHub, hosted on GitHub pages, and made runnable/editable in the p5 editor without needing to constantly update each sketch.

Glitch has a "[repo import](http://help-center.glitch.me/help/import-git/)" feature, but I don't know if that makes sense to import an entire repo. Every time you press "play" it would bring you back to the main "index.html" which would feel weird.

The functionality I'm imagining is most similar to the way that Google colab will start a new colab notebook from a GitHub URL, [described here](https://tensorchiefs.github.io/dl_course_2018/co.html).

**Comment by catarak** [comment]
i have also run into this issue wrt teaching! one approach to this problem i've been thinking about is API access, i.e. #541 and #542. however, this issue is making me think there's another approach, which would be tight github integration. i like this idea of seamlessly linking everything, though i have no idea how technically feasible that would be. another approach would be a simple way to upload to and from github. a good first step for this would be some investigation.

**Comment by Fupete** [comment]
A kudo from me also on a "git" repository sync feature! Your thread is from November I don't know if there is some updates however my 10 cents: I taught a uni/student workshop yesterday about P5 & Open Source and we shared with the students a vision in which this new incredible sketch editor could be in sync when needed via a classic add/commit/push/pull git workflow with a git repo somewhere (github, bitbucket, ...). Could be quite usefull in many ways for sure.

**Comment by kylemath** [comment]
agreed, this would be an excellent feature, I collaborate with artists who want to see deployed version but also play with the code in editor , currently I have to push to repo, copy code into live editor,

**Comment by slowizzm** [comment]
🥺🥺🥺

**Comment by Ajaya1000** [comment]
Hey, @catarak I think that it needs an extensive one-to-one conversation in order to solve this issue. So can I mention this issue in my GSoC proposal?

**Comment by alin256** [comment]
A very much wanted feature for me as well. 

Is there any progress?

**Comment by RandomGamingDev** [comment]
Agreed. This would be really helpful when dealing with things like merging projects with multiple edits on different pages or even with different people and for its version control.

**Comment by FYCodeLab** [comment]
I agree it would be a great addition !
